server/game/cards/02_SHD/events/MaKlounkee.ts

```diff
@@ -0,0 +1,33 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { EventCard } from '../../../core/card/EventCard';
+import { RelativePlayer, Trait, WildcardCardType } from '../../../core/Constants';
+
+export default class MaKlounkee extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '0931441928',
+            internalName: 'ma-klounkee',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Return a friendly non-leader Underworld unit to its owner\'s hand',
+            targetResolver: {
+                controller: RelativePlayer.Self,
+                cardTypeFilter: WildcardCardType.NonLeaderUnit,
+                cardCondition: (card, context) => card.hasSomeTrait(Trait.Underworld),
```

Comment:
nit: since we don't use the context variable you can change it to `_context` to signify that